dffml/config/config.py

```diff
@@ -71,3 +75,58 @@ def __call__(self) -> BaseConfigLoaderContext:
         config_path = tuple(config_path)
         # Load the file
         return config_path, await parsers[filetype].loadb(path.read_bytes())
+
+    @classmethod
+    async def load_file(
+        cls,
+        parsers: Dict[str, "BaseConfigLoader"],
+        exit_stack: contextlib.AsyncExitStack,
+        path: pathlib.Path,
+        *,
+        base_dir: Optional[pathlib.Path] = None,
+    ) -> Dict:
+        async def _get_config(temp_filepath):
+            if not isinstance(temp_filepath, pathlib.Path):
+                temp_filepath = pathlib.Path(temp_filepath)
+            _, loaded = await BaseConfigLoader.load_single_file(
+                parsers, exit_stack, temp_filepath, base_dir=base_dir
+            )
+            return loaded
+
+        if len(path.suffixes) >= 2 and path.suffixes[-2] == ".dirconf":
+            dir_name = path.parts[-1].split(".")[0]
+            dir_path = os.path.join(*(path.parts[:-1] + (dir_name,)))
+
+            temp_conf_dict = {dir_name: dir_path}
+            conf_dict = await _get_config(path)
+            explored = explore_directories(temp_conf_dict)
+            explored = await nested_apply(explored, _get_config)
+            conf_dict.update(explored[dir_name])
+        else:
+            conf_dict = await _get_config(path)
+        return conf_dict
```

Comment:
If we don't maintain the return of `config_path, ` we'll break the multicomm config loading stuff. Let's make sure we're still returning it for now.

Reply:
Done